src/core/client/stream/tabs/Comments/Comment/CommentContainer.tsx

```diff
@@ -82,438 +89,421 @@ interface Props {
-  showEditDialog: boolean;
-  editable: boolean;
-  showReportFlow: boolean;
-}
-
-export class CommentContainer extends Component<Props, State> {
-  private uneditableTimer: any;
-
-  public state = {
-    showReplyDialog: false,
-    showEditDialog: false,
-    editable: this.isEditable(),
-    showReportFlow: false,
-  };
+export const CommentContainer: FunctionComponent<Props> = ({
+  className,
+  collapsed,
+  comment,
+  disableReplies,
+  hideAnsweredTag,
+  hideModerationCarat,
+  highlight,
+  indentLevel,
+  localReply,
+  onRemoveAnswered,
+  settings,
+  showConversationLink,
+  hideReportButton,
+  story,
+  toggleCollapsed,
+  eventEmitter,
+  setCommentID,
+  viewer,
+  showAuthPopup,
+}) => {
+  const [showReplyDialog, setShowReplyDialog] = useState(false);
+  const [
+    showEditDialog,
+    setShowEditDialog,
+    toggleShowEditDialog,
+  ] = useToggleState(false);
+  const [showReportFlow, , toggleShowReportFlow] = useToggleState(false);
+  const handleShowConversation = useCallback(
+    (e: MouseEvent) => {
+      ViewConversationEvent.emit(eventEmitter, {
+        commentID: comment.id,
+        from: "COMMENT_STREAM",
+      });
+      e.preventDefault();
 
-  constructor(props: Props) {
-    super(props);
-    if (this.isEditable()) {
-      this.uneditableTimer = this.updateWhenNotEditable();
-    }
-  }
+      void setCommentID({ id: comment.id });
 
-  public componentWillUnmount() {
-    clearTimeout(this.uneditableTimer);
-  }
+      return false;
+    },
+    [eventEmitter, comment, setCommentID]
+  );
 
-  private isEditable() {
-    const isMyComment = !!(
-      this.props.viewer &&
-      this.props.comment.author &&
-      this.props.viewer.id === this.props.comment.author.id
-    );
-    const banned = Boolean(
-      this.props.viewer &&
-        this.props.viewer.status.current.includes(GQLUSER_STATUS.BANNED)
-    );
-    const suspended = Boolean(
-      this.props.viewer &&
-        this.props.viewer.status.current.includes(GQLUSER_STATUS.SUSPENDED)
-    );
-    return (
-      !banned &&
-      !suspended &&
-      isMyComment &&
-      !!this.props.comment.editing.editableUntil &&
-      isBeforeDate(this.props.comment.editing.editableUntil)
-    );
-  }
-
-  private toggleReplyDialog = () => {
-    if (this.props.viewer) {
-      this.setState((state) => {
-        if (!state.showReplyDialog) {
-          ShowReplyFormEvent.emit(this.props.eventEmitter, {
-            commentID: this.props.comment.id,
-          });
-        }
-        return {
-          showReplyDialog: !state.showReplyDialog,
-        };
+  const openEditDialog = useCallback(() => {
+    if (viewer) {
+      ShowEditFormEvent.emit(eventEmitter, {
+        commentID: comment.id,
       });
+      setShowEditDialog(true);
     } else {
-      void this.props.showAuthPopup({ view: "SIGN_IN" });
+      void showAuthPopup({ view: "SIGN_IN" });
     }
-  };
+  }, [viewer, eventEmitter, comment, showAuthPopup]);
```

Comment:
Possible optimisation: `comment.id` instead of `comment`. 
Turn viewer into a boolean before and use that in deps:

```
const isLoggedIn = Boolean(viewer);
const openEditDialog = useCallback(() => {
  if (isLoggedIn) {
   ...
}, [isLoggedIn, ...]);
```